ParseStarterProject/src/com/parse/starter/LoginActivity.java

```diff
@@ -40,7 +40,7 @@ private void login() {
         String username = usernameEditText.getText().toString().trim();
         String password = passwordEditText.getText().toString().trim();
 
-        if(username.length() != 0 && password.length() != 0) {
+        if(username.length() != 0 && password.length() != 0) {  //requires user and pass not empty
```

Comment:
The problem is that login exits successfuly when it was not successful.  The parent needs to know that.

```Java
// ...
public void onCreate(Bundle savedInstanceState) {
	// ...
	loginButton.setOnClickListener(new View.OnClickListener() {
		@Override
		public void onClick(View v) {
			try {
				login();
			} catch (SomeSortOfException e) {
				Toast.makeText(LoginActivity.this, e.getMessage());
				return;
			}
		// ...
	}
}
// ...
```